integration/setup/teardown.go

```diff
@@ -52,11 +54,6 @@ func teardown(ctx context.Context, config Config) error {
 		}
 	}
 
-	{
-		// TODO there should be error handling for the framework teardown.
-		config.Host.Teardown()
-	}
```

Comment:
`Teardown` only calls a helm command that fails because the used name is wrong. I added the deletion of the Vault release above using the key functions. I would then also drop `Teardown` from the host framework in `e2e-harness`.